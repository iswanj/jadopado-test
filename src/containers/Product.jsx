import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import { setState } from '../actions';

import styles from 'product.css';

import ProductItem from './components/ProductItem';

export default class Product extends PureComponent {
  render() {
    return (
      <div className={styles.mainContainer}>
        <section className={styles.products}>
          <ProductItem />
        </section>
      </div>
    );
  }
}

// Connect to Redux store
function mapStateToProps(state, ownProps) {
  return {

  };
}

export const ProductContainer = connect(
  mapStateToProps,
  { setState }
)(Product);

/*
var reg = /\#.* /gi

var str = "asdfkadsf #xxxxx bb";

var rpText = reg.exec(str);

var test = str.replace(reg,"<a href='#'>"+rpText[0].replace(/ /g,'')+"</a> ");

console.log(test);
*/
