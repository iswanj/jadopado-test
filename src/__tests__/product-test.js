jest.mock('react-dom');
import React from 'react';
import Product from '../containers/Product';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('Product page renders correctly', () => {
  const router = {
    params: {
      id: 1
    }
  };

  const products = {
    1: {
      "id": 1,
      "title": "Leaf iPhone Case Hard Plastic",
      "owner": {
        "username": "tracymacgrady",
        "image": "https://randomuser.me/api/portraits/men/13.jpg"
      },
      "images": [
        "leaf-iphone-case.jpg"
      ],
      "price": 450,
      "distance": 500,
      "likes": 32,
      "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit",
      "tages": ["iphone", "cases", "mobile_phones", "macbookpro"],
      "comments": [
        {
          "id": 1,
          "username": "jevonjans",
          "image": "https://randomuser.me/api/portraits/men/98.jpg",
          "comment": "Adhuc liberavisse cu pro. Ex eam mutat ancillae noluisse. Ius ad autem aliquid vulputate"
        },
        {
          "id": 2,
          "username": "robynrihanna",
          "image": "https://randomuser.me/api/portraits/women/28.jpg",
          "comment": "Te nec soleat graece commune, cu pri ferri verear dignissim, case wisi ad vim. Modo iusto aperiam eu eos, etiam convenire assueverit cum no."
        }
      ]
    }
  };

  const tree = renderer.create(
    <Product products={products} router={router} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
