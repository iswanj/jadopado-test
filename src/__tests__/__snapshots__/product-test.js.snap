exports[`test Product page renders correctly 1`] = `
<div
  className={undefined}>
  <section
    className={undefined}>
    <div
      className={undefined}>
      <div
        className={undefined}>
        <div
          className={undefined}>
          <img
            alt="User Pic"
            className={undefined}
            src="https://randomuser.me/api/portraits/men/13.jpg" />
          <span
            className={undefined}>
            tracymacgrady
          </span>
        </div>
        <div
          className={undefined}>
          <span>
            500
            m
          </span>
           away
        </div>
      </div>
      <div
        className={undefined}
        onClick={[Function]}
        style={
          Object {
            "width": "1024px",
          }
        }>
        <div
          className={undefined}
          style={
            Object {
              "backgroundImage": "url(/images/leaf-iphone-case.jpg)",
              "width": "1024px",
            }
          } />
        <div
          className={undefined}>
          <div
            className={undefined}>
            <p>
              Leaf iPhone Case Hard Plastic
            </p>
            <p>
              <strong>
                AED 
                450
              </strong>
            </p>
          </div>
          <div
            className={undefined}>
            <i
              className="  " />
            <i
              className="  " />
          </div>
        </div>
      </div>
      <div
        className={undefined}>
        <i
          className="  " />
         
        32
         likes
      </div>
      <div
        className={undefined}>
        <p
          className={undefined}>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit
          <span
            className={undefined}>
             
            #iphone 
            #cases 
            #mobile_phones 
            #macbookpro 
          </span>
        </p>
      </div>
      <div
        className={undefined}>
        <div
          className={undefined}>
          <div
            className={undefined}>
            <img
              alt="jevonjans"
              src="https://randomuser.me/api/portraits/men/98.jpg" />
          </div>
          <div
            className={undefined}>
            <span
              className={undefined}>
              jevonjans
            </span>
             
            Adhuc liberavisse cu pro. Ex eam mutat ancillae noluisse. Ius ad autem aliquid vulputate
          </div>
        </div>
        <div
          className={undefined}>
          <div
            className={undefined}>
            <img
              alt="robynrihanna"
              src="https://randomuser.me/api/portraits/women/28.jpg" />
          </div>
          <div
            className={undefined}>
            <span
              className={undefined}>
              robynrihanna
            </span>
             
            Te nec soleat graece commune, cu pri ferri verear dignissim, case wisi ad vim. Modo iusto aperiam eu eos, etiam convenire assueverit cum no.
          </div>
        </div>
      </div>
      <div
        className={undefined}>
        <input
          className={undefined}
          placeholder="Write comment"
          type="text" />
        <button
          className={undefined}
          onClick={[Function]}>
          <i
            className="  " />
        </button>
      </div>
    </div>
  </section>
  <div
    className={undefined}>
    <div
      className={undefined}>
      <a
        className={undefined}
        onClick={[Function]}
        style={Object {}}>
        <i
          className="  " />
      </a>
    </div>
    <div
      className={undefined}>
      Photos
    </div>
    <div
      className={undefined} />
  </div>
</div>
`;
